onnxruntime/core/common/threadpool.cc

```diff
@@ -24,28 +26,27 @@
 namespace onnxruntime {
 
 namespace concurrency {
-#ifdef USE_EIGEN_THREADPOOL
 
 // TODO: This is temporarily taken from Eigen until we upgrade its version.
 // Barrier is an object that allows one or more threads to wait until
 // Notify has been called a specified number of times.
 class Barrier {
  public:
   Barrier(unsigned int count) : state_(count << 1), notified_(false) {
```

Comment:
"TODO: This is temporarily taken from Eigen until we upgrade its version."
I have updated Eigen. So, do we still need this class?

Reply:
We also need a Barrier for std threapool. Barrier in std is experimental: https://en.cppreference.com/w/cpp/experimental/barrier . I prefer to keep it for both eigen and std threadpool.

Reply:
Not in this PR

---
In reply to: [277931462](https://github.com/Microsoft/onnxruntime/pull/891#discussion_r277931462) [](ancestors = 277931462)